integration/test/Testlib/ModService.hs

```diff
@@ -353,44 +347,54 @@ withProcess resource overrides service = do
 
   startNginzLocalIO <- lift $ appToIO $ startNginzLocal resource
 
+  serviceHandleMVar <- liftIO newEmptyMVar
   let initProcess = case (service, cwd) of
         (Nginz, Nothing) -> startNginzK8s domain sm
         (Nginz, Just _) -> startNginzLocalIO
         _ -> do
           config <- getConfig
           tempFile <- writeTempFile "/tmp" (execName <> "-" <> domain <> "-" <> ".yaml") (cs $ Yaml.encode config)
           (_, Just stdoutHdl, Just stderrHdl, ph) <- createProcess (proc exe ["-c", tempFile]) {cwd = cwd, std_out = CreatePipe, std_err = CreatePipe}
-          let prefix = "[" <> execName <> "@" <> domain <> "] "
-          let colorize = fromMaybe id (lookup execName processColors)
-          void $ forkIO $ logToConsole colorize prefix stdoutHdl
-          void $ forkIO $ logToConsole colorize prefix stderrHdl
+          (out1, out2) <- mkChans stdoutHdl
+          (err1, err2) <- mkChans stderrHdl
+          void $ forkIO $ logChanToConsole execName domain out1
+          void $ forkIO $ logChanToConsole execName domain err1
+          putMVar serviceHandleMVar (out2, err2, ph)
```

Comment:
As discussed it is probably best to do this without the MVar. I will push the code in a bit.